crypto/bio/bss_dgram.c

```diff
@@ -641,7 +645,10 @@ static long dgram_ctrl(BIO *b, int cmd, long num, void *ptr)
                 perror("getsockopt");
                 ret = -1;
             } else if (sizeof(sz.s) != sizeof(sz.i) && sz.i == 0) {
-                OPENSSL_assert(sz.s <= sizeof(struct timeval));
+                if (!ossl_assert(sz.s <= sizeof(struct timeval))) {
+                    ret = -1;
+                    break;
+                }
```

Comment:
Same as previous comment.

On side note, it makes as much sense to get rid of perror, effectively with same rationale.